src/main/java/hudson/tasks/Mailer.java

```diff
@@ -509,20 +546,25 @@ public FormValidation doSendTestMail(
                 @QueryParameter boolean useSsl, @QueryParameter String smtpPort, @QueryParameter String charset,
                 @QueryParameter String sendTestMailTo) throws IOException, ServletException, InterruptedException {
             try {
+                final Jenkins jenkins = Jenkins.getInstance();
+                if (jenkins == null) {
+                    throw new IOException("Jenkins instance is not ready");
+                }
+                
                 if (!useSMTPAuth) {
                     smtpAuthUserName = null;
                     smtpAuthPasswordSecret = null;
                 }
                 
                 MimeMessage msg = new MimeMessage(createSession(smtpServer, smtpPort, useSsl, smtpAuthUserName, smtpAuthPasswordSecret));
-                msg.setSubject(Messages.Mailer_TestMail_Subject(++testEmailCount), charset);
-                msg.setText(Messages.Mailer_TestMail_Content(testEmailCount, Jenkins.getInstance().getDisplayName()), charset);
-                msg.setFrom(StringToAddress(adminAddress, charset));
+                msg.setSubject(Messages.Mailer_TestMail_Subject(incrementTestEmailCount()), charset);
```

Comment:
I wonder what findbugs complains about = as the new code isn't thread safe, just like the old code...